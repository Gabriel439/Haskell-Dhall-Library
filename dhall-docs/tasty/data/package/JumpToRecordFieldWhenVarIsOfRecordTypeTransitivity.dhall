{-|
On this example, `dhall-docs` will detect that the variable `a` is bounded
to a record literal and it will be aware that it uses dot-syntax.

Transitivity is respected when the variable is assigned using let-bindings.
For instance, if the user clicks the `y` from the `ax.y.z`, it will jump to
the definition of `x` on the record literal.

_(the code is the same from [](./JumpToRecordFieldWhenVarIsOfRecordTypeWithDotSyntax.dhall))_
-}
let a = { x.y.z = 1 }

let ax = a.x
let axy = a.x.y
let axyz = a.x.y.z

in ax.y.z + axy.z + axyz
