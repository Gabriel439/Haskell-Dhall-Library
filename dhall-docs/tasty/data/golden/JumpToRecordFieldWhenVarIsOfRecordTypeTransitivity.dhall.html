<!DOCTYPE HTML>
<html
  ><head
    ><title
    >/JumpToRecordFieldWhenVarIsOfRecordTypeTransitivity.dhall</title
    ><link href="index.css" type="text/css" rel="stylesheet"
    /><link href="https://fonts.googleapis.com/css2?family=Fira+Code:wght@400;500;600;700&amp;family=Lato:ital,wght@0,400;0,700;1,400&amp;display=swap"
            type="text/css" rel="stylesheet"
    /><script src="index.js" type="text/javascript"
    /><meta charset="UTF-8"/></head
  ><body
    ><div class="nav-bar"
      ><img src="dhall-icon.svg" class="dhall-icon"
      /><p class="package-title"
      >test-package</p
      ><div class="nav-bar-content-divider"
      /><a id="switch-light-dark-mode" class="nav-option"
      >Switch Light/Dark Mode</a></div
    ><div class="main-container"
      ><h2 class="doc-title"
        ><span class="crumb-divider"
        >/</span
        ><a href="index.html"
        >test-package</a
        ><span class="crumb-divider"
        >/</span
        ><span href="index.html" class="title-crumb"
        >JumpToRecordFieldWhenVarIsOfRecordTypeTransitivity.dhall</span></h2
      ><a data-path="/JumpToRecordFieldWhenVarIsOfRecordTypeTransitivity.dhall"
          class="copy-to-clipboard"
        ><i
          ><small
          >Copy path to clipboard</small></i></a
      ><br
      /><div class="doc-contents"
        ><p
          >On this example, <code
          >dhall-docs</code
          > will detect that the variable <code
          >a</code> is bounded
to a record literal and it will be aware that it uses dot-syntax.</p
        >
<p
          >Transitivity is respected when the variable is assigned using let-bindings.
For instance, if the user clicks the <code
          >y</code
          > from the <code
          >ax.y.z</code
          >, it will jump to
the definition of <code
          >x</code> on the record literal.</p
        >
<p
          ><em
            >(the code is the same from <a href="./JumpToRecordFieldWhenVarIsOfRecordTypeWithDotSyntax.dhall"
            >./JumpToRecordFieldWhenVarIsOfRecordTypeWithDotSyntax.dhall</a>)</em></p>
</div
      ><h3
      >Source</h3
      ><div class="source-code"
        ><pre
          >{-|<br
          />On this example, `dhall-docs` will detect that the variable `a` is bounded<br
          />to a record literal and it will be aware that it uses dot-syntax.<br
          /><br
          />Transitivity is respected when the variable is assigned using let-bindings.<br
          />For instance, if the user clicks the `y` from the `ax.y.z`, it will jump to<br
          />the definition of `x` on the record literal.<br
          /><br
          />_(the code is the same from &lt;./JumpToRecordFieldWhenVarIsOfRecordTypeWithDotSyntax.dhall&gt;)_<br
          />-}<br
          />let <span id="var11-5" data-variable="var11-5"
                      class="variable-decl"
          >a</span
          > = { <span id="var11-11" data-variable="var11-11"
                      class="variable-decl"
          >x</span
          >.<span id="var11-13" data-variable="var11-13"
                  class="variable-decl"
          >y</span
          >.<span id="var11-15" data-variable="var11-15"
                  class="variable-decl"
          >z</span
          > = 1 }<br
          /><br
          />let <span id="var13-5" data-variable="var13-5"
                      class="variable-decl"
          >ax</span
          > = <a href="#var11-5" data-variable="var11-5" class="variable-use"
          >a</a
          >.<a href="#var11-11" data-variable="var11-11" class="variable-use"
          >x</a
          ><br
          />let <span id="var14-5" data-variable="var14-5"
                      class="variable-decl"
          >axy</span
          > = <a href="#var11-5" data-variable="var11-5" class="variable-use"
          >a</a
          >.<a href="#var11-11" data-variable="var11-11" class="variable-use"
          >x</a
          >.<a href="#var11-13" data-variable="var11-13" class="variable-use"
          >y</a
          ><br
          />let <span id="var15-5" data-variable="var15-5"
                      class="variable-decl"
          >axyz</span
          > = <a href="#var11-5" data-variable="var11-5" class="variable-use"
          >a</a
          >.<a href="#var11-11" data-variable="var11-11" class="variable-use"
          >x</a
          >.<a href="#var11-13" data-variable="var11-13" class="variable-use"
          >y</a
          >.<a href="#var11-15" data-variable="var11-15" class="variable-use"
          >z</a
          ><br
          /><br
          />in <a href="#var13-5" data-variable="var13-5"
                  class="variable-use"
          >ax</a
          >.<a href="#var11-13" data-variable="var11-13" class="variable-use"
          >y</a
          >.<a href="#var11-15" data-variable="var11-15" class="variable-use"
          >z</a
          > + <a href="#var14-5" data-variable="var14-5" class="variable-use"
          >axy</a
          >.<a href="#var11-15" data-variable="var11-15" class="variable-use"
          >z</a
          > + <a href="#var15-5" data-variable="var15-5" class="variable-use"
          >axyz</a
          ><br/></pre></div></div></body></html>